torch/_subclasses/fake_tensor.py

```diff
@@ -589,13 +582,22 @@ def __init__(
         # normalize cuda device.
         if device.type == "cuda" and device.index is None:
             device = torch.device(f"cuda:{torch.cuda.current_device()}")
-        self.fake_device = device
-        self.fake_mode = fake_mode
-        self.constant = constant
+        self.fake_device = device  # type: ignore[attr-defined]
+        self.fake_mode = fake_mode  # type: ignore[attr-defined]
+        self.constant = constant  # type: ignore[attr-defined]
         if FakeTensorConfig.debug:
             import traceback
 
-            self._debug_trace = traceback.extract_stack()
+            self._debug_trace = traceback.extract_stack()  # type: ignore[attr-defined]
+        return self
+
+    # In some circumstances, a conventional torch.Tensor constructor
+    # will get rewritten to call into FakeTensor.  We must provide an
+    # __init__ method that can accept the Python interpreters initialization
+    # in such a situation; we must also be able to handle direct fake
+    # tensor construction via FakeTensor()
+    def __init__(self, *args, **kwargs):
```

Comment:
I'm a bit surprised, aren't the arguments here exactly the same as the ones passed to `__new__` ?
In which case is the `__init__` called but not the `__new__` ?

Reply:
Referencing https://github.com/pytorch/pytorch/pull/92754#discussion_r1084271697 the situation is

```
with FakeTensorMode():
    torch.Tensor([1, 2, 3])
```

this becomes

```
x = torch.Tensor.__new__([1, 2, 3])
x.__init__([1, 2, 3])
```

due to the fake tensor mode, x is a FakeTensor, and thus we are calling `__init__` on fake tensor with the "wrong" arguments (not the fake tensor initialization arguments).